Sources/Image.swift

```diff
@@ -15,40 +15,74 @@ import UIKit
 public struct Image: NSViewRepresentable {
     let imageContainer: ImageContainer
     let onCreated: ((ImageView) -> Void)?
+    @Binding var isVideoLooping: Bool
+    var onVideoFinished: (() -> Void)?
+    var restartVideo: Bool = false
 
     public init(_ image: NSImage) {
         self.init(ImageContainer(image: image))
     }
 
-    public init(_ imageContainer: ImageContainer, onCreated: ((ImageView) -> Void)? = nil) {
+    public init(_ imageContainer: ImageContainer,
+                isVideoLooping: Binding<Bool> = .constant(true),
+                onCreated: ((ImageView) -> Void)? = nil) {
         self.imageContainer = imageContainer
+        self._isVideoLooping = isVideoLooping
         self.onCreated = onCreated
     }
 
     public func makeNSView(context: Context) -> ImageView {
         let view = ImageView()
+        if view.isVideoLooping != isVideoLooping {
+            view.isVideoLooping = isVideoLooping
+        }
+        view.onVideoFinished = onVideoFinished
         onCreated?(view)
         return view
     }
 
     public func updateNSView(_ imageView: ImageView, context: Context) {
+        if imageView.isVideoLooping != isVideoLooping {
+            imageView.isVideoLooping = isVideoLooping
+        }
+        if restartVideo {
+            imageView.videoReplay()
```

Comment:
The `videoReplay` call is missing. I'm going to remove it temporarily for the upcoming release.

Reply:
I changed it to `restartVideo()`. I think that's what you meant?